tensorflow/core/kernels/tensor_to_hash_bucket_op_gpu.cu.cc

```diff
@@ -0,0 +1,120 @@
+/* Copyright 2021 The TensorFlow Authors. All Rights Reserved.
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+==============================================================================*/
+
+#if GOOGLE_CUDA
+
+#define EIGEN_USE_GPU
+#include "third_party/eigen3/unsupported/Eigen/CXX11/Tensor"
+#include "tensorflow/core/framework/register_types.h"
+#include "tensorflow/core/kernels/tensor_to_hash_bucket_op.h"
+#include "third_party/farmhash_gpu/src/farmhash_gpu.h"
+#include "tensorflow/core/util/gpu_kernel_helper.h"
+
+namespace tensorflow {
+
+namespace {
+
+constexpr int kSharedMemBufferSizePerThread = 64;
```

Comment:
OOC why does it need to be this large?  Since `T` is `int64` or narrower, the stringified version will only have 20 chars at most right?

Reply:
I set it to 64 is because originally I planned to support other data types and formats (like integers with user-specified width/padding, floats with different precisions). But since we only support plain integers at this moment, I changed it to 32.

Reply:
Why not `19` instead of `32`?

Reply:
Sure. Reset it to 20, since both max uint64 and min int64 require 20 chars.